src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayController.cs

```diff
@@ -93,6 +97,81 @@ public async Task<IActionResult> ViewApplication(Guid applicationId)
             }
         }
 
+        [HttpGet("/Roatp/Gateway/{applicationId}/ConfirmOutcome")]
+        public async Task<IActionResult> ConfirmOutcome(Guid applicationId)
+        {
+            var application = await _applyApiClient.GetApplication(applicationId);
+            if (application is null)
+            {
+                return RedirectToAction(nameof(NewApplications));
+            }
+
+            var username = _contextAccessor.HttpContext.User.UserDisplayName();
+            var viewModel = await _orchestrator.GetConfirmOverviewViewModel(new GetApplicationOverviewRequest(applicationId, username));
+
+            if (viewModel.ReadyToConfirm)
+            {
+                return View("~/Views/Roatp/Apply/Gateway/ConfirmOutcome.cshtml", viewModel);
+            }
+            else
+            {
+                return Redirect($"/Roatp/Gateway/{applicationId}");
+            }
+
+        }
+
+        [HttpPost("/Roatp/Gateway/{applicationId}/ConfirmOutcome")]
+        public async Task<IActionResult> EvaluateConfirmOutcome(RoatpGatewayApplicationViewModel viewModel)
+        {
+            var validationResponse = await _validator.Validate(viewModel);
+
+            if (validationResponse.Errors != null && validationResponse.Errors.Any())
+            {
+                var username = _contextAccessor.HttpContext.User.UserDisplayName();
+                var errorViewModel = await _orchestrator.GetConfirmOverviewViewModel(new GetApplicationOverviewRequest(viewModel.ApplicationId, username));
+                if(errorViewModel != null)
+                {
+                    errorViewModel.ErrorMessages = validationResponse.Errors;
```

Comment:
Could the validationResponse be passed into the orchestrator so the building of the viewmodel is done in one place?